tests/dopamine/replay_memory/circular_replay_buffer_test.py

```diff
@@ -638,6 +645,35 @@ def testLoad(self):
     self.assertEqual(memory.add_count, self._test_add_count)
     self.assertAllClose(memory.invalid_range, self._test_invalid_range)
 
+  def testMemoryLocksItself(self):
+    """Tests that adding an element to the buffer blocks the lock."""
+    memory = _create_dummy_memory()
+    memory._lock = mock.MagicMock()
+
+    # Add one element.
+    memory.add((1, 2), 0, 0, False)
+    # Check that buffer contains one element.
+    self.assertEqual(memory.add_count, 1)
+    # Check that the lock went throughh the proper lock/unlock process.
+    memory._lock.__enter__.assert_called_once()
+    memory._lock.__exit__.assert_called_once()
+
+  def testAddMethodIsBlockedWhenLockIsHeld(self):
+    """Tests that when the lock is blocked elements cannot be added."""
+    memory = _create_dummy_memory()
+    memory._lock = mock.MagicMock()
+    memory._lock.__enter__ = mock.Mock(
+        side_effect=ValueError('Lock is locked.'))
+    with self.assertRaisesRegexp(ValueError, 'Lock is locked.'):
+      memory.add((1, 2), 0, 0, False)
+    # Check that the second element was not added.
+    self.assertEqual(memory.add_count, 0)
+
+  def testMemoryHasAProperLock(self):
+    """Tests that lock attribute is initialized properly."""
+    memory = _create_dummy_memory()
+    self.assertTrue(hasattr(memory, '_lock'))
```

Comment:
if you follow my suggestion above, you could test different names here, as well as ensuring that the attribute doesn't exist when `lock_attribute_name == None`

Reply:
Since I opted for option 2, I added a test to checks that the provided lock object in the `OutOfGraphhReplayBuffer` init is actually set as `_lock`.
In the case where locking is disable, that value should be `None`